test/active_record_doctor/detectors/missing_unique_indexes_test.rb

```diff
@@ -30,6 +30,20 @@ def test_missing_unique_index_on_functional_index
     run_detector
   end
 
+  def test_validates_multiple_attributes
+    create_table(:users) do |t|
+      t.string :email
+      t.string :ref_token
+      t.index :email, unique: true
+    end.create_model do
+      validates :email, :ref_token, uniqueness: true
+    end
+
+    assert_problems(<<~OUTPUT)
+      add a unique index on users(ref_token) - validating uniqueness in the model without an index can lead to duplicates
```

Comment:
I think the test case should remove the unique index on `email` and test whether both `email` and `ref_token` are reported -- otherwise, we'd let an implementation testing only the first attribute pass through the tests.